### PR TITLE
STAR-15 Edit question

### PR DESCRIPTION
**Pull request by farzaneh-haghani**
- Added `EditQuestionFrom`
- Added an `Endpoint` for updating
- minor bug fixed

### CODE REVIEW

File: server/src/controllers/questionsController.ts [hunk 1]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
...
+  }
+
+  try {
+    const updateQuestionQuery = await editQuestion(questionId, question);
+    logger.info({
+      message: "updateQuestionQuery data",
+      value: updateQuestionQuery
+    });
+    const data = updateQuestionQuery[0];
+    res.status(200).json({ message: "success", data: data });
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:338]:
```suggestion
    res.status(200).json(data);
```

Let's stick to the same pattern as the others, we can check on the frontend for response.ok

File: server/src/controllers/questionsController.ts [hunk 2]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
       .json({ error: "Error Adding Your Comment to the Database" });
   }
 };
+
+export const editQuestionHandler = async (req: Request, res: Response) => {
+  const questionId = Number(req.params.id);
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:319]:
```suggestion
  const questionId = parseInt(req.params.id);
  logger.info({
    message: "editQuestionHandler questionId",
    value: questionId
  });
```

lets change to using `parseInt`` as you suggested
and we need to add a logger here

File: server/src/controllers/questionsController.ts [hunk 3]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
...
+
+  if (!questionId) {
+    res.status(400).json({ error: "Question Id was not provided" });
+  }
+
+  const question = req.body.questionContent;
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:325]:
```suggestion
  const question = req.body.questionContent;
  logger.info({
    message: "editQuestionHandler question",
    value: question
  });
```
we need to add logger here

File: server/src/controllers/questionsController.ts [hunk 4]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
...
+
+  if (!question) {
+    res.status(400).json({ error: "Question was not provided" });
+  }
+
+  try {
+    const updateQuestionQuery = await editQuestion(questionId, question);
+    logger.info({
+      message: "updateQuestionQuery data",
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:334]:
```suggestion
      message: "updateQuestionQuery query",
```

File: server/src/controllers/questionsController.ts [hunk 5]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
...
+      value: updateQuestionQuery
+    });
+    const data = updateQuestionQuery[0];
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:337]:
```suggestion

    const data = updateQuestionQuery[0];
    logger.info({  
      message: "editQuestionHandler data",  
      value: data  
    }); 

```
add spaces and logger

File: server/src/controllers/questionsController.ts [hunk 6]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
       .json({ error: "Error Adding Your Comment to the Database" });
   }
 };
+
+export const editQuestionHandler = async (req: Request, res: Response) => {
+  const questionId = Number(req.params.id);
+
+  if (!questionId) {
+    res.status(400).json({ error: "Question Id was not provided" });
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:322]:
```suggestion
    res.status(400).json({ error: "Invalid Question ID Provided" });
```

File: server/src/controllers/questionsController.ts [hunk 7]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
...
+  }
+
+  const question = req.body.questionContent;
+
+  if (!question) {
+    res.status(400).json({ error: "Question was not provided" });
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:328]:
```suggestion
    res.status(400).json({ error: "No Question on the Request Body" });
```

File: client/src/components/AddCommentForm.jsx [hunk 8]
```diff
@@ -19,8 +19,8 @@ import {
 } from "../themes/ConsistentStyles";
 
 const AddCommentForm = ({ questionId, answerId, setShowAddCommentForm }) => {
-  console.log("AddCommentForm questionId:", questionId);
-  console.log("AddCommentForm answerId:", answerId);
+  // console.log("AddCommentForm questionId:", questionId);
+  // console.log("AddCommentForm answerId:", answerId);
```
Comment (bazmurphy) [inline client/src/components/AddCommentForm.jsx:23]:
```suggestion
```
we don't need these now

File: client/src/components/EditQuestionForm.jsx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+        throw new Error("Failed to Edit question");
+      }
+      // const data = await response.json();
+      // console.log("Edit data:", data);
+      setIsEditing(false);
+      setStatus({
+        submitting: false,
+        error: false,
+        success: true,
+        message: "Your Edit successfully saved!",
```
Comment (bazmurphy) [inline client/src/components/EditQuestionForm.jsx:88]:
```suggestion
        message: "Your Question was successfully Edited!",
```

File: client/src/components/EditQuestionForm.jsx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+        error: undefined,
+      });
+    } catch (error) {
+      console.error(error);
+      setStatus({
+        submitting: false,
+        error: true,
+        success: false,
+        message:
+          "There was an error to send your Editing Question to the Server",
```
Comment (bazmurphy) [inline client/src/components/EditQuestionForm.jsx:101]:
```suggestion
          "There was an error sending your Edited Question to the Server",
```

**Review by bazmurphy — CHANGES_REQUESTED** [review]

File: server/src/controllers/questionsController.ts [hunk 11]
```diff
@@ -313,3 +314,29 @@ export const createCommentHandler = async (req: Request, res: Response) => {
...
+  try {
+    const updateQuestionQuery = await editQuestion(questionId, question);
+    logger.info({
+      message: "updateQuestionQuery data",
+      value: updateQuestionQuery
+    });
+    const data = updateQuestionQuery[0];
+    res.status(200).json({ message: "success", data: data });
+  } catch (error) {
+    res.status(500).json({ error: "Error Editing Your Question" });
```
Comment (bazmurphy) [inline server/src/controllers/questionsController.ts:340]:
```suggestion
    res.status(500).json({ error: "Server Error" });
```

**Comment by bazmurphy** [comment]
Good Job @farzaneh-haghani 
Just a few suggested changes and then good to go ⭐ 😎

File: client/src/components/AddCommentForm.jsx [hunk 12]
```diff
@@ -35,27 +35,27 @@ const AddCommentForm = ({ questionId, answerId, setShowAddCommentForm }) => {
...
+
     setComment((prevComment) => {
       return {
         ...prevComment,
         content: event.target.value,
         error:
-          event.target.value.length < 10 || event.target.value.length > 500
-            ? true
-            : false,
+          value.trim().length < 10 || value.trim().length > 500 ? true : false,
```
Comment (jarrodruby) [inline client/src/components/AddCommentForm.jsx:45]: Can we add https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/minlength the minlength and maxlength fields

**Review by bazmurphy — APPROVED** [review]
Good Job ⭐ 😎 

We can make another ticket to fix all of the different forms we have built with input restrictions etc.

**Pull request merged**